app/controllers/TaxReturnReceivedController.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import common.SessionValues
+import config.{AppConfig, ErrorHandler}
+import controllers.predicates.AuthorisedAction
+import controllers.predicates.TaxYearAction.taxYearAction
+import javax.inject.{Inject, Singleton}
+import models.TaxReturnReceivedModel
+import org.joda.time.format.DateTimeFormat
+import org.joda.time.{DateTime, DateTimeZone}
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import services.AuthService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.TaxReturnReceivedView
+import utils.SessionDataHelper
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class TaxReturnReceivedController @Inject()(val authorisedAction: AuthorisedAction,
+                                            authService: AuthService,
+                                            val taxReturnReceivedView: TaxReturnReceivedView,
+                                            errorHandler: ErrorHandler,
+                                            implicit val appConfig: AppConfig,
+                                            implicit val mcc: MessagesControllerComponents,
+                                            implicit val ec: ExecutionContext) extends FrontendController(mcc) with I18nSupport with SessionDataHelper {
+
+  val timeStamp: String = DateTimeFormat.forPattern("d MMMM YYYY").print(DateTime.now(DateTimeZone.UTC))
+
+  def show(taxYear: Int): Action[AnyContent] = (authorisedAction andThen taxYearAction(taxYear)).apply { implicit user =>
+
+    val summaryDataReceived: Option[TaxReturnReceivedModel] = getSessionData[TaxReturnReceivedModel](SessionValues.SUMMARY_DATA)
+
+    summaryDataReceived match {
+      case Some(summaryData) =>
+        Ok(taxReturnReceivedView(summaryData, user.isAgent, taxYear, timeStamp))
+      case _ =>
+        errorHandler.handleError(INTERNAL_SERVER_ERROR)
```

Comment:
I'm not sure displaying a 500 page here would be good as the 'Try again later' content we have implies it might work later on which in this case it won't if there isn't session data. 

I'm thinking we log that they've tried to access the page without the data being there and redirect to overview page, what you think?

Reply:
Had a chat with Duane about this, the issue being is that we don't really have any way of telling the user if their submission has failed otherwise

Reply:
In the future there would be the actual declare your income tax return API call as part of the show in this controller which would handle the failure scenarios of the users submission failing.

The session data not being there can only happen if the user has bookmarked the page without the session data or the data has expired. Does beg the question of whether we need to keep the result for longer than in session. But in these cases I think a redirect works

Reply:
Not really a comment on here but when we get round to doing the actual declare call, I think it makes sense to check the sessionData is there before doing the declare call too.

It not being present can only mean it's not been added in v&c and the user is not in a position to declare